packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js

```diff
@@ -14,28 +14,107 @@ import noMatchesFor from '@salesforce/label/c.quantic_NoMatchesFor';
 import collapseFacet from '@salesforce/label/c.quantic_CollapseFacet';
 import expandFacet from '@salesforce/label/c.quantic_ExpandFacet';
 
+/** @typedef {import("coveo").SearchEngine} SearchEngine */
+/** @typedef {import("coveo").CategoryFacet} CategoryFacet */
+/** @typedef {import("coveo").CategoryFacetState} CategoryFacetState */
+/** @typedef {import("coveo").CategoryFacetValue} CategoryFacetValue */
+
+/**
+ * A facet is a list of values for a certain field occurring in the results, ordered using a configurable criteria (e.g., number of occurrences).
+ * A `QuanticCategoryFacet` displays a facet of values in a browsable, hierarchical fashion.
+ * @category LWC
+ * @example
+ * <c-quantic-category-facet field="geographicalhierarchy" label="Country" engine-id={engineId}></c-quantic-category-facet>
+ */
 export default class QuanticCategoryFacet extends LightningElement {
-  /** @type {import("coveo").CategoryFacetState} */
-  // @ts-ignore TODO: Check CategoryFacetState typing and integration with LWC/Quantic
-  @track state = {
-    values: [],
-    parents: [],
-  };
-
-  /** @type {string} */
+  /**
+   * The ID of the engine instance with which to register.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /** 
+   * An unique ID used to identify the facet instance.
+   * Defaults to given facet label.
+   * @api
+   * @type {string}
+   * @defaultValue `(field)`
```

Comment:
Do you mean that the parentheses are part of the ID? I.e., the ID is `(field)`, not `field`?

Reply:
they are not. i didnt know how to specify this. by default it uses the value specified as a field. should i just put `field` ?

Reply:
anything is possible :rainbow: